### PR TITLE
GoPro Synchronization Tool

### PR DESCRIPTION
**Pull request by VivaSuresh**
Created Gopro sync tool and would like for it to be added to the fishsense-tools repository.

### CODE REVIEW

File: Gopro-Sync/goprosync.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+def gopro_sync(left, right, trimmed_left, trimmed_right):
+
+    # Create a temporary directory to store the audio files
+    
+    try:
+        os.mkdir(".temp")
+    except FileExistsError:
+        pass
+    
+    curr_directory = Path(os.getcwd()).joinpath(".temp")
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Simpler to do this:
```
curr_directory = Path('.temp')
curr_directory.mkdir(exists_ok=True)
```

File: Gopro-Sync/goprosync.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    right_wav.close()
+    
+    Path.unlink(left_audio_path)
+    Path.unlink(right_audio_path)
+    
+    # Only delete the directory if it's not already empty
+    try:
+        Path.rmdir(curr_directory)
+    except Exception:
+        pass
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Simpler to use `shutil`
```
shutil.rmtree(curr_directory)
```

File: Gopro-Sync/goprosync.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    left_trimmed = args.out.joinpath("left_trimmed.mp4")
+    right_trimmed = args.out.joinpath("right_trimmed.mp4")
+
+    gopro_sync(left=args.left, right=args.right, trimmed_left=left_trimmed, trimmed_right=right_trimmed)
+
+if __name__ == '__main__':
+    main()
+
+
+
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Extra newlines

File: .gitignore
Comment (ntlhui) [inline .gitignore:1]: Use https://github.com/UCSD-E4E/python-repo-example/blob/main/.gitignore and add all appropriate additional ignores

File: Gopro-Sync/goprosync.py
Comment (ntlhui) [inline Gopro-Sync/goprosync.py:1]:
```
fishsense-tools\Gopro-Sync> pylint .\goprosync.py
************* Module goprosync
goprosync.py:6:40: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:14:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:19:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:49:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:53:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:56:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:63:59: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:92:0: C0301: Line too long (102/100) (line-too-long)
goprosync.py:93:0: C0301: Line too long (137/100) (line-too-long)
goprosync.py:98:40: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:115:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:129:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:133:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:137:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:143:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:147:0: C0303: Trailing whitespace (trailing-whitespace)
goprosync.py:151:0: C0301: Line too long (104/100) (line-too-long)
goprosync.py:157:0: C0305: Trailing newlines (trailing-newlines)
goprosync.py:1:0: C0114: Missing module docstring (missing-module-docstring)
goprosync.py:11:0: C0116: Missing function or method docstring (missing-function-docstring)
goprosync.py:11:0: R0914: Too many local variables (37/15) (too-many-locals)
goprosync.py:60:11: W0718: Catching too general exception Exception (broad-exception-caught)
goprosync.py:88:4: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
goprosync.py:11:0: R0915: Too many statements (59/50) (too-many-statements)
goprosync.py:114:0: C0116: Missing function or method docstring (missing-function-docstring)
goprosync.py:128:8: R1722: Consider using 'sys.exit' instead (consider-using-sys-exit)
goprosync.py:132:8: R1722: Consider using 'sys.exit' instead (consider-using-sys-exit)
goprosync.py:136:8: R1722: Consider using 'sys.exit' instead (consider-using-sys-exit)
goprosync.py:142:8: R1722: Consider using 'sys.exit' instead (consider-using-sys-exit)
goprosync.py:146:8: R1722: Consider using 'sys.exit' instead (consider-using-sys-exit)
goprosync.py:5:0: C0411: standard import "import os" should be placed before "import numpy as np" (wrong-import-order)
goprosync.py:7:0: C0411: standard import "from pathlib import Path" should be placed before "import numpy as np" (wrong-import-order)
goprosync.py:8:0: C0411: standard import "import argparse" should be placed before "import numpy as np" (wrong-import-order)
goprosync.py:3:0: W0611: Unused matplotlib.pyplot imported as plt (unused-import)

-----------------------------------
Your code has been rated at 6.34/10
```

File: Gopro-Sync/goprosync.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    
+    # Make sure that input files are mp4s
+
+    if (args.left.as_posix()[-4:] != ".mp4" and args.left.as_posix()[-4:] != ".MP4"):
+        print("Ensure that left file is an mp4")
+        exit()
+    
+    if (args.right.as_posix()[-4:] != ".mp4" and args.right.as_posix()[-4:] != ".MP4"):
+        print("Ensure that right file is an mp4")
+        exit()
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Please use https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix

File: Gopro-Sync/goprosync.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+import stumpy
+import os
+from moviepy.editor import VideoFileClip 
+from pathlib import Path
+import argparse
+
+
+def gopro_sync(left, right, trimmed_left, trimmed_right):
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Add type hints:
for example, `gopro_sync(left: Path, right: Path)`

File: Gopro-Sync/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+This tool is to sychronize two different gopro videos by using an impulse noise, usually a clap. The tool uses this clap to trim the videos. 
+
+In this folder, there are two different files.
+One is a gopro sychronization tool to be used in the command line, and the other is a jupyter notebook that shows plots for better understanding.
+
+gopro_sync tool usage:
+python goprosync.py --left PATH_TO_LEFT_MP4 --right PATH_TO_RIGHT_MP4 --out PATH_TO_OUTPUT_DIRECTORY
```
Comment (ntlhui) [inline Gopro-Sync/README.md:7]: Use the output from `python .\goprosync.py --help`

File: Gopro-Sync/goprosync.py [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        print("Left path is not a file.")
+        exit()
+    
+    elif not args.right.is_file():
+        print("Right path is not file")
+        exit()
+    
+    elif not args.out.is_dir():
+        print("Out path is not a directory")
+        exit()
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: Maybe you should just create the directory?

File: Gopro-Sync/goprosync.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+
+def gopro_sync(left, right, trimmed_left, trimmed_right):
+
+    # Create a temporary directory to store the audio files
+    
+    try:
+        os.mkdir(".temp")
+    except FileExistsError:
+        pass
```
Comment (ntlhui) [inline Gopro-Sync/goprosync.py]: What happens if creating `.temp` fails because the system is unable to create `.temp`?

**Review by ntlhui — CHANGES_REQUESTED** [review]
I'm getting some weird test case behavior.  I am using this video: https://photos.app.goo.gl/5Z6Ctz9Bzv1D8RibA.  I them use ffmpeg to trim the first 2 seconds from the video (resulting is the last 8 seconds of the original).  I then use that for left and right.  The output videos have a different aspect ratio (view is squashed), and I only have the data after the clap.  Is this the intended behavior?  It would seem better for this to only trim out the non-common time segment.

**Comment by VivaSuresh** [comment]
Fixed errors. Creating a new pull request.

**Pull request closed without merging**